### ISSUE TITLE
[unbound-method] Wrapping method in arrow function doesn't satisfy linter when passed as argument

### ISSUE THREAD

**Issue by melink14** [typescript-eslint/typescript-eslint#3448]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": "error",
  }
}
```

```TS
class TestClass {
  myMethod() {
    const x = 1 + 1;
  }
  constructor() {
    document.addEventListener('onload', () => this.myMethod);
  }
}
```

**Expected Result**

There should be no error since the class method is wrapped in an arrow function as per the docs.

**Actual Result**

> 6:47  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`. 

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.22.1` |
| `@typescript-eslint/parser`        | `4.22.1` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.26.0` |
| `node`                             | `14.16.1` |

**Comment by melink14** [comment]
Ah, nevermind, I'm not invoking my callback so it's acting as the return value which could still be dangerous. Confirmed that when I correctly called it then everything worked.
